addons/website_payment/models/payment_transaction.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, fields, models
+
+
+class PaymentTransaction(models.Model):
+    _inherit = "payment.transaction"
+
+    is_donation = fields.Boolean(string="Is donation", help="Is the payment a donation")
+
+    def _finalize_post_processing(self):
+        for tx in self:
+            if tx.is_donation:
+                tx._send_donation_email()
```

Comment:
```suggestion
        for tx in self.filtered('is_donation'):
            tx._send_donation_email()
```

Reply:
Feel free to apply or discard as my suggestion is likely less efficient than a `if` inside a `for`. However, this only becomes an issue when traversing recordsets of thousands of records, which is clearly not going to be the case here.